### PR TITLE
Fix broken `--kubecontext`

### LINKED ISSUE #1351: `--kubecontext` in any `subctl` command doesn't work
`kubecontexts` was added in 0.9 but it broke `kubecontext` for any other command that uses it.

Specifically, these lines [1] can't work because at this pint in the code the flags are just being initialized and have no values yet.

On a grander scheme, I think `kubecontexts` is wrong and if users want to loop over several contexts, they can call the command several time with a different `kubecontext` each time.

For commands that require two contexts (or allow 1 or 2 contexts) an additional flag can be used which allows to specify the additional context explicitly and clearly.

Hence I propose:
1. Remove the `kubecontexts` flag
2. Fix back `kubecontext` to working
3. For command requiring another context (currently only `verify` AFAICT), add a flag `othercontext` to specify the other context. This flag can be verified to be set and proper, and the users can be warned appropriately if it's unset.
4. As per `kubectl` and other CLI tools, if no context is specified we should honor the `current-context` as set in the kubeconfig file.

[1] https://github.com/submariner-io/submariner-operator/pull/1223/files#diff-8a58275c37bca775e6338952d361c5a6f84bfc2956093290bb661ed575c7c9d0R64-R67

### PR DESCRIPTION
**Pull request by mkolesnik**
Reverted it to use it's own variable (which works fine).

Also renamed the function names so its easy to distinguish between the
different functionality.

Resolves: #1351 

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: pkg/subctl/cmd/root.go [hunk 1]
```diff
@@ -62,21 +62,26 @@ func Execute() error {
...
 	cmd.PersistentFlags().StringVar(&kubeConfig, "kubeconfig", "", "absolute path(s) to the kubeconfig file(s)")
-	cmd.PersistentFlags().StringSliceVar(&kubeContexts, "kubecontext", nil, "kubeconfig context to use")
-	if len(kubeContexts) > 0 {
-		kubeContext = kubeContexts[0]
-	}
 }
 
-func addKubecontextsFlag(cmd *cobra.Command) {
+// addKubecontextFlag adds a "kubeconfig" flag and a single "kubecontext" flag that can be used once and only once
+func addKubecontextFlag(cmd *cobra.Command) {
```
Comment (tpantelis) [inline pkg/subctl/cmd/root.go]:
```suggestion
func addKubeContextFlag(cmd *cobra.Command) {
```

File: pkg/subctl/cmd/root.go [hunk 2]
```diff
@@ -62,21 +62,26 @@ func Execute() error {
...
+	addKubeconfigFlag(cmd)
+	cmd.PersistentFlags().StringVar(&kubeContext, "kubecontext", "", "kubeconfig context to use")
+}
+
+// addKubecontextMultiFlag adds a "kubeconfig" flag and a "kubecontext" flag that can be specified multiple times (or comma separated)
+func addKubecontextMultiFlag(cmd *cobra.Command) {
```
Comment (tpantelis) [inline pkg/subctl/cmd/root.go]:
```suggestion
func addKubeContextMultiFlag(cmd *cobra.Command) {
```

**Review by tpantelis — DISMISSED** [review]
couple nits

**Comment by mkolesnik** [comment]
Be sure to squash when merging

**Review by dfarrell07 — APPROVED** [review]

**Comment by vthapar** [comment]
This will reaquire changes to documentation for `subctl gather` which uses `--kubecontexts`.

**Review by skitt — APPROVED** [review]

**Comment by mkolesnik** [comment]
> This will reaquire changes to documentation for `subctl gather` which uses `--kubecontexts`.

WDYM? That flag still exists - https://github.com/submariner-io/submariner-operator/pull/1355/files#diff-8a58275c37bca775e6338952d361c5a6f84bfc2956093290bb661ed575c7c9d0R82

**Comment by vthapar** [comment]
> > This will reaquire changes to documentation for `subctl gather` which uses `--kubecontexts`.
> 
> WDYM? That flag still exists - https://github.com/submariner-io/submariner-operator/pull/1355/files#diff-8a58275c37bca775e6338952d361c5a6f84bfc2956093290bb661ed575c7c9d0R82

Nvm, I missed that.

**Pull request merged**